### PR TITLE
QA reporting enhancements

### PR DESCRIPTION
**Pull request by TylerHendrickson**
This PR tweaks various parts of the repo involved in the QA phase of the CI workflow, with particular focus on reinforcing existing (lint) checks related to code quality and consistency.

Summary of changes:
1. Fix an issue in the CI workflow where ESLint checks for `web/src` were skipped if ESLint failed for `api/src`. We want ESLint to run for both sides, even if one of them fails. This also fixes an issue where ESLint checks were incorrectly reporting as passing.
2. Implement suggestions for avoiding rate-limiting on TFLint plugin download by ensuring they are being cached and reused across workflow runs.
3. Added a quick guide in `docs/` for developer reference when dealing with QA check failures.
4. Resolved all current ESLint errors and warnings (only necessary for `api/src`).

### CODE REVIEW

File: api/src/lib/aws.ts [hunk 1]
```diff
@@ -61,7 +62,7 @@ function getS3Client() {
 export function uploadWorkbook(
   upload: CreateUploadInput,
   uploadId: number,
-  body: any
+  body: StreamingBlobPayloadInputTypes
```
Comment (TylerHendrickson) [inline api/src/lib/aws.ts:65]: Using types from [`@smithy/types`](https://www.npmjs.com/package/@smithy/types) as a more descriptive alternative. 

This way, type-checkers can determine whether `body` values are compatible with the S3 PutObject operation.

File: api/src/lib/aws.ts [hunk 2]
```diff
@@ -70,8 +71,8 @@ export function uploadWorkbook(
 async function sendPutObjectToS3Bucket(
   bucketName: string,
   key: string,
-  body: any
-) {
+  body: StreamingBlobPayloadInputTypes
```
Comment (TylerHendrickson) [inline api/src/lib/aws.ts:74]: Same as [above](https://github.com/usdigitalresponse/cpf-reporter/pull/55/files#r1443395603).

File: api/src/lib/aws.ts [hunk 3]
```diff
@@ -115,12 +116,11 @@ export async function s3PutSignedUrl(
   })
   return url
 }
+
 /**
  *  This function is a wrapper around the getSignedUrl function from the @aws-sdk/s3-request-presigner package.
  *  Exists to organize the imports and to make it easier to mock in tests.
  */
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (TylerHendrickson) [inline api/src/lib/aws.ts:123]: This function's signature doesn't appear to have any unused vars, so removed this pragma.

File: api/src/lib/aws.ts [hunk 4]
```diff
@@ -143,8 +143,7 @@ function getSQSClient() {
   return sqs
 }
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (TylerHendrickson) [inline api/src/lib/aws.ts:146]: This function's signature doesn't appear to have any unused vars, so removed this pragma.

File: api/src/lib/aws.ts [hunk 5]
```diff
@@ -154,7 +153,6 @@ async function sendSqsMessage(queueUrl: string, messageBody: any) {
   )
 }
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (TylerHendrickson) [inline api/src/lib/aws.ts:157]: This function's signature doesn't appear to have any unused vars, so removed this pragma.

File: api/src/lib/aws.ts [hunk 6]
```diff
@@ -143,8 +143,7 @@ function getSQSClient() {
   return sqs
 }
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-async function sendSqsMessage(queueUrl: string, messageBody: any) {
+async function sendSqsMessage(queueUrl: string, messageBody: unknown) {
```
Comment (TylerHendrickson) [inline api/src/lib/aws.ts:146]: This post makes a compelling argument for preferring `unknown` over `any` for typing values that are passed through to `JSON.stringify()`, so decided to go with it.

**Review by as1729 — APPROVED** [review]

**Pull request merged**